### DISCUSSION TITLE
✅ 🔗 Add a new Nested Off Canvas widget (Duplicate of #6225) - [ED-13937]

### DISCUSSION THREAD

**Discussion by Michaelgimii** [elementor/elementor#21769] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Since most website visitors are mobile, it is important to create a standard mobile menu to optimize the website for mobile users. **Now, we can create a mobile menu using popup + icon widget.** But this is not an optimal solution.
Most importantly, the problem is that we put the open mobile menu button in the header, but to edit the menu content, we have to edit the popup. Therefore, to create a mobile menu in the current layout, we need to switch between the header and the popup(menu content).
**But with the addition of this widget, we add the widget in the header and edit and control its content there. This is the power of nested elements!**

### Describe the solution you'd like

Please add a new widget (nested mobile menu). This widget will be very useful. In addition to the mobile menu, it is very useful in displaying product filters on mobile phones. This widget will be only in the form of an icon and text(image below).

![a](https://user-images.githubusercontent.com/88034902/227768664-2922a0bb-ecaf-4c00-9937-db7ca390736a.png)

After clicking on it, its content will be displayed. Just like the Mega menu widget (pictured below).

![b](https://user-images.githubusercontent.com/88034902/227768121-1869d459-e216-4b7c-b968-5abb694f9856.png)

rival page builder will add this feature soon. Please [see here](https://prnt.sc/c4HD6rhAtKne). It is clear from the number of votes that it is very popular. **(Of course, I'm not sure if they'll add it like I'm saying)**

Anyway, this is a super important feature and I hope you put it in the roadmap. Some very important features that are needed for nested menu mobile :

1. Control the width of mobile menu content (like popup)
2. Entrance Animation and Exit Animation
3. Custom icon for close button
4. Overlay (like popup)
5. Ability to control icons and text in normal, hover and active modes : https://prnt.sc/jGCrq0suhLh7

Hi @avivu and @arielk, I hope you see this request❤ thanks

**Comment by gheorghe9512** [comment]
I agree with @Michaelgimii, It is something of a priority I would say because I have a lot of sites that do them and I have to use popup which is horrible, or with CSS and container to get a menu.

**Comment by Joliamendi** [comment]
this is fabulous! I now create menu using popup for mobile. But it really bothers me. Also to display product filter on mobile phone.

I would like to add another request for this great feature. I hope you consider it.

It would be great if you let us create a custom close button in this widget, for example using an icon widget. Just like now we can create custom close button for popup using icon widget or other widgets.

https://user-images.githubusercontent.com/103376243/227783785-f071f42c-db9e-4cc3-923e-44c52f8f42e5.mp4

**Comment by SolunaCG** [comment]
I just opened another issue when I found this! This is exactly what I´d love to see! Mobile menus are always very basic in terms of customisation and since we now (almost) have the great new Mega Menu it´d be amazing to have an option for mobile menus, too! The suggestion I made in my previously opened thread was to just completely get rid of the "responsive settings breakpoint" since this already was a problem with the new "Tabs"-widget, as mentioned here:

https://github.com/elementor/elementor/issues/7742

I really wonder why this option exists at all. Having the complete freedom of designing menus for every breakpoint and not only for Desktop and Tablet would be awesome! I don´t know why these limitations were made in the first place. The new "Tabs" widget was pretty much unusable for me because of how it looked on mobile and it´s the same with the new "Mega Menu". So either a new "Nested Menu Mobile" as mentioned above would be great or (since we already have the widgets for Tabs and now the Mega Menu) an option to disable that "responsive settings breakpoint" completely and let us design the mobile menu instead so that we have only one widget (the new Mega Menu) to create a menu for all devices and not separate ones for 1.:mobile and 2.: all the other devices.

My menu works great on Tablet (I just tried it out, not polished though) but it breaks on Mobile as you can see because of that annoying setting.

![Bildschirm­foto 2023-03-28 um 16 11 35](https://user-images.githubusercontent.com/114682524/228278520-647bc865-9674-4209-a619-2d351b8a9e10.png)
![Bildschirm­foto 2023-03-28 um 16 21 52](https://user-images.githubusercontent.com/114682524/228278524-5da635ee-203c-4b96-a942-c70d17fb677c.png)

**Comment by nicholaszein** [comment]
Hi @Michaelgimii,

I think what you're requesting here is more of a "Nested Offcanvas". Isn't that a more accurate description of what you are requesting?

**Reply by Michaelgimii** [reply]
Hi @nicholaszein 

Yes, exactly. "Nested Offcanvas" is a better name for it. I hope it is in your roadmap.

**Comment by 79ho3ein** [comment]
This widget is really needed👌.

**Comment by fabigrafo** [comment]
+1!!

**Comment by hitsgee** [comment]
Crocoblock have a free plugin that adds this functionality and is pretty easy to use. I have it on numerous sites some without any other crocoblock plugin installed and no issues.

https://github.com/Crocoblock/jet-offcanvas-column

I would still love for this to be core Elementor though in the future.

**Comment by SolunaCG** [comment]
Even though this is a little bit offtopic I´d still like to post it here cause it has to do with mobile menu in general: 

You know what I don´t understand? Please don´t get me wrong. I´m not upset or anything. I would just like to understand why (apparently) so many Elementor users don´t seem to need a good mobile menu. I´m not talking about the "Nested Off Canvas menu" but the standard Elementor mobile menu in general. Imo the "Nested Off Canvas" is more specific than the standard one. However, from what I´ve seen in the Elementor discussions on github (unfortunately) there doesn´t seem to be a high demand for a better mobile menu (which is very weird to me). 

So what I´d like to know is: Why? I mean... Do you guys really think that the current Elementor mobile menu is good?? Or do you use 3rd party plugins just for that like Unlimited Elements, Happy Addons, etc..?
As someone who has been creating websites for many years with Elementor for many clients I keep getting the feedback from these clients that they don´t like the standard Elementor mobile menu at all. It´s not just a slight "dislike", it´s more that they keep telling me how bad it is in general. 
I´m not going to mention all the things that are missing in the current mobile menu again because I already created a discussion for that here: #23634 

So yea, please let me know if you really like it how it currently is or if you´re using 3rd party addons or what´s your way to go there. 
Because I still can´t imagine that literally ALL of my clients are SO picky when it comes to a mobile menu. I guess they´ve just noticed as well how bad the Elementor one looks like.

**Comment by nicholaszein** [comment]
Hello there!

We have great news! 🙌 

### 📢 We're happy to announce that this feature was released with Elementor Pro v3.22.0! 🥳 

✅ Feel free to check it out and update your plugin to the new version!

> _Check the changelog for more information:_
> [Changelog of Elementor and Elementor Pro](https://elementor.com/help/changelog-of-elementor/)

Cheers 🥂
